### ISSUE TITLE
missing real cipsoft monsters behavior

### ISSUE THREAD

**Issue by renanluis90** [joseluis2g/my-nostalrius#22]
when the monster can not attack the player, he is behaving in a wrong way.
the delay in the monster's walking, is very slow...

**Here the actually monsters behavior movie.**
https://streamable.com/y7xzc

we can clearly see that the monsters have a bigger delay in the updatelook ...

**Here the test of ReaLOTS monsters behavior movie.**
https://streamable.com/wobml

they are faster, with updatelook

**Comment by joseluis2g** [comment]
I think my changes fucked it up, they are facing on every step

**Comment by renanluis90** [comment]
yes, i saw... because in ReaLOTS, theys "random" the updatelook yes? and theys are more faster

**Comment by joseluis2g** [comment]
thats related to nextDanceStepRound
try playing with this
```
nextDanceStepRound = OTSYS_TIME() + 200 + getStepDuration();
```
like remove the 200

**Comment by joseluis2g** [comment]
@emil92b could you test the new changes
https://github.com/joseluis2g/my-nostalrius/commit/6a2b8b6aa866e0fee7c1831c68f3f3eed3ab1f27

**Comment by emil92b** [comment]
the previous direction change worked better, think you broke it

also the dancestep thing dont work either

**Comment by joseluis2g** [comment]
Can you test both and do some images about it? I think I do not understand what you want.
Also with danceStep, I just did the changes and monsters seems to walk faster when they fleeing/walking around

**Comment by Stoudemire** [comment]
https://streamable.com/wobml

así es como debería funcionar 
implemente el código pero ahora caminan demasiado rápido y no se voltean a mirar el target (player)

**Comment by ghost** [comment]
i noticed if you run with
`nextDanceStepRound = OTSYS_TIME() + 100 + getStepDuration();`

monster on same floor dance every 1sec

but the monsters on other floors dance every 2 seconds and also sometimes every 1 second, so that is an issue, they should dance every 1 second there, not randomly 2-1

**Comment by joseluis2g** [comment]
but the ground has his own speed reduction or walk cost, can you test it?

**Comment by ghost** [comment]
i tested it here on experimental
https://i.imgur.com/BRreFES.jpg (1sec)
https://i.imgur.com/uSdvLXB.jpg (1sec)

**Comment by joseluis2g** [comment]
I need to test this a few, I do not know where to start.

**Comment by renanluis90** [comment]
Creature.cpp? getStepDuration function

**Comment by renanluis90** [comment]
nextDanceStepRound = OTSYS_TIME() + 760;

**Comment by joseluis2g** [comment]
@renanluis90 have you tested it? it is directly related to this dancestep but I have to do more tests because if you use a lower number there is no check of system time and they walk too fast

**Comment by renanluis90** [comment]
I will upload a video
